djangocms_snippet/admin.py

```diff
@@ -98,19 +98,17 @@ def has_delete_permission(self, request, obj=None):
 
     def has_change_permission(self, request, obj=None):
         """
-        Return edit mode if current user is the author, otherwise display snippet in read only mode
+        Return read only mode if q parameter is in URL, otherwise return edit mode
         """
         if obj and djangocms_versioning_enabled:
-            created_by_id = self.get_version(obj).created_by_id
```

Comment:
We need to keep this logic because it is for version locking. 

We may end up in the situation where the parameter has not been supplied but the version is not locked to the current user. 

May need to verify this logic and double check it to as without version locking I think that anyone can change it.